### PR TITLE
CircleCI: run evm2wasm tests with shared Hera

### LINKED ISSUE #305: CI: Test evm2wasm.js with Aleth binaries
Convert the CircleCI job "evm2wasm-tests" to be like "linux-clang-shared-*" (or include the tests in this job), i.e. build hera to a shared library and use downloaded binaries of Aleth instead of building it from source.

### PR DESCRIPTION
**Pull request by axic**
Fixes #305.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -275,12 +216,35 @@ jobs:
       - *build
       - *save-deps-cache
 
+  linux-shared-evm2wasm:
```
Comment (chfast) [inline circle.yml]: Don't you want to merge this tests with `linux-*-shared-release` or `linux-*-shared-coverage`?

**Comment by axic** [comment]
> Does this include tests with evm2wasm.js?

We do not use evm2wasm.js anymore. It is too slow.

File: circle.yml (hunk 1 above)
Comment (axic) [reply circle.yml]: I don't want to include it in coverage.

Comment (chfast) [reply circle.yml]: So add it to  linux-clang-shared-release

**Comment by axic** [comment]
@chfast rebased, can you take this over?

**Comment by chfast** [comment]
Yes

File: circle.yml [hunk 2]
```diff
@@ -137,6 +123,20 @@ defaults:
         testeth --version
         testeth -t GeneralStateTests/stEWASMTests -- --testpath tests --vm ~/build/src/libhera.so --singlenet Byzantium
 
+  fetch-evm2wasm-tests: &fetch-evm2wasm-tests
+    run:
+      name: "Fetch evm2wasm tests"
+      command: |
+        git clone https://github.com/ethereum/tests
```
Comment (axic) [inline circle.yml]: This is wrong, should clone `ewasm/tests` with `ewasm-tests` branch.

**Comment by chfast** [comment]
I will do it today.

**Comment by chfast** [comment]
Included in `linux-clang-shared-release` but fails on some tests.

**Comment by axic** [comment]
Can you just use the regular tests repo for evm2wasm? Don't want to spend any time on this.

**Comment by chfast** [comment]
Sure.

**Comment by chfast** [comment]
No clue why it does not work.

**Comment by axic** [comment]
> No clue why it does not work.

I think it may be the entire problem we're having, that `aleth 1.4.x` is somehow broken.

The only difference in this PR is that the old version runs off master of aleth.

**Comment by chfast** [comment]
Ok, it works now.

**Comment by chfast** [comment]
Can we merge this?

**Pull request merged**